### ISSUE TITLE
0.19.0: false positive on MD044

### ISSUE THREAD

**Issue by okalachev** [igorshubovych/markdownlint-cli#68]
Starting with 0.19.1 markdownlint-cli began incorrectly finding improper names ([`MD044`](https://github.com/DavidAnson/markdownlint/blob/master/doc/Rules.md#md044)) in URLs.

Text:

```markdown
Read more in the PX4 docs: https://docs.px4.io/v1.9.0/en/config/gyroscope.html.
```

`markdownlint.json` (fragment):

```javascript
    "MD044": {
        "names": [
            "HTTPS"
            // ...
        ],
        "code_blocks": false
    }
```

Output:

```
docs/en/calibration.md:28 MD044/proper-names Proper names should have the correct capitalization [Expected: HTTPS; Actual: https]
```

**Comment by DavidAnson** [comment]
The above is a bare URL; is it still a problem if you put angle brackets around that? (On my phone right now, can’t check.)

Also, version 0.19.0, right? 0.19.1 doesn’t exist yet.

**Comment by okalachev** [comment]
Yes, this happens exactly with bare URLs. On the previous version (0.18.0) this checks did not fail. Bracketed urls are not affected.

Yes, the version 0.19.0.

**Comment by okalachev** [comment]
Maybe the problem is with `markdownlint` itself though. I only use it through `markdownlint-cli`.

**Comment by DavidAnson** [comment]
Yes, this issue is with the core library. Still my bug, though. :)

**Comment by DavidAnson** [comment]
May I ask why this is a bare URL? If you bracket it (there’s a rule for that with auto-fix!) and the problem goes away, is that sufficient?

**Comment by okalachev** [comment]
Sometimes bare URLs are just more convenient, for example, when you want to make the site name where you point the reader more obvious/emphasised.

Compare:

I think the best repository on GitHub is located on https://github.com/igorshubovych/markdownlint-cli.

>I think the best repository on GitHub is [markdownlint-cli](https://github.com/igorshubovych/markdownlint-cli).

>I wouldn't say that the first is always better, I just say these are different, and sometimes the first is preferred.

To fix it, I should do something like:

```markdown
[https://docs.px4.io/v1.9.0/en/config/compass.html](https://docs.px4.io/v1.9.0/en/config/compass.html)
```

While I have a lot of bare URLs in my documentation, and this is redundant and not cool. Much easer to change the version to the previous one, and wait until it's fixed.

**Comment by DavidAnson** [comment]
Makes sense, thanks. I’m hopeful this is straightforward to fix.

**Comment by DavidAnson** [comment]
This is a great bug - not at all what I expected. I will have a fix in the core library tomorrow and an updated release in a few days.

**Comment by DavidAnson** [comment]
Here’s the fix: https://github.com/DavidAnson/markdownlint/commit/c44c7600edaf6a73db523ced1fcbb350bd7495b7

I will release this as a patch update to the core library, so it should not be necessary to publish a new release of the CLI. Running `npm update` should be enough.

**Comment by DavidAnson** [comment]
`markdownlint` 0.17.2 is released - please do a package update and confirm the new version is being used - your issue should be fixed.

**Comment by okalachev** [comment]
Seem to work, thanks!
